### PR TITLE
[docs] Add doc for fingerprint runtime version and CI/CD (beta)

### PR DESCRIPTION
**Pull request by wschurman**
# Why

This adds documentation for the new `fingerprint` runtime version policy for expo-updates and how to use it to do CI/CD.

Closes ENG-11157.

# How

Add docs. Note that more changes are coming to the github action we provide to be up to date with this doc, but since it's in beta I think it's okay to order them this way to start dogfooding.

# Test Plan

Start docs server locally, go to page, proofread.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/eas-update/runtime-versions.mdx [hunk 1]
```diff
@@ -100,7 +100,27 @@ It's important to know that this policy does require the developer to manage the
 
 Also, if you select a different native version between Android and iOS, you'll end up with builds and updates with separate runtime versions. When you publish an update, EAS CLI will detect that two updates are needed, and it will create two updates with the required runtime versions and publish them separately.
 
-### `"fingerprintExperimental"` runtime version policy (experimental)
+### `"fingerprint"` runtime version policy
```
Comment (quinlanj) [inline docs/pages/eas-update/runtime-versions.mdx:103]: Why not just leave it as `fingerprintExperimental`? Is it so people who were relying on it in sdk 50 and below will need to make an active decision to switch?

**Review by quinlanj — APPROVED** [review]

File: docs/pages/eas-update/runtime-versions.mdx (hunk 1 above)
Comment (wschurman) [reply docs/pages/eas-update/runtime-versions.mdx:103]: It's a valid question. Your guess is correct. They're incompatible with each other, so we don't want people who used `fingerprintExperimental` to accidentally be upgraded to `fingerprint` automatically.

File: docs/pages/eas-update/continuous-deployment.mdx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - (optional) Submit to app stores for truly continuous deployment. This has not yet been tested and is not recommended.
+4. Run `eas update` to publish an update for each commit.
+
+### GitHub Actions
+
+Expo provides a [GitHub Action](https://github.com/expo/expo-github-action/tree/main/continuous-deploy-fingerprint) for performing the steps above within a GitHub Actions workflow.
+
+## Debugging fingerprint inconsistencies
+
+If you notice different fingerprints being generated across different machines or environments, it may mean that there are unanticipated files being included in the hash calculation. `@expo/fingerprint` has a predetermines set of files to include/exclude for hash calculation, but often your project setup may require additional excludes. For projects that contain native code directories (`./android` and/or `./ios`) this is more common.
```
Comment (amandeepmittal) [inline docs/pages/eas-update/continuous-deployment.mdx:33]:
```suggestion
If you notice different fingerprints being generated across different machines or environments, it may mean that unanticipated files are being included in the hash calculation. `@expo/fingerprint` has a predetermines set of files to include/exclude for hash calculation, but often your project setup may require additional excludes. For projects that include native directories (**android** and **ios**) this is more common.
```

Bold file and directory as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text)

File: docs/pages/eas-update/continuous-deployment.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+We provide tools for identifying which files are causing fingerprint inconsistencies and mechanisms to exclude those files from fingerprint calculations for your project.
+
+To identify differences in fingerprint on different machines or environments:
+- When running fingerprint generation commands on each machine/environment (`npx expo-updates fingerprint:generate`), pass `--debug` flag.
+- Diff outputs from those command runs to determine files causing the difference. These tools may be helpful:
+    - https://jsonformatter.org/json-pretty-print to format the output.
+    - https://www.jsondiff.com/ to compare the output and identify the files causing the discrepancies.
```
Comment (amandeepmittal) [inline docs/pages/eas-update/continuous-deployment.mdx:41]:
```suggestion
    - [JSON Pretty Print](https://jsonformatter.org/json-pretty-print ) to format the output.
    -[JSON Diff](https://www.jsondiff.com/) to compare the output and identify the files causing the discrepancies.
```

File: docs/pages/eas-update/continuous-deployment.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+To exclude files causing the differences, add them to `.fingerprintignore` as described in the [`@expo/fingerprint` README](https://www.npmjs.com/package/@expo/fingerprint).
```
Comment (amandeepmittal) [inline docs/pages/eas-update/continuous-deployment.mdx:43]:
```suggestion
To exclude files causing the differences, add them to the **.fingerprintignore** file as described in the [`@expo/fingerprint` README](https://www.npmjs.com/package/@expo/fingerprint).
```

File: docs/pages/eas-update/runtime-versions.mdx [hunk 5]
```diff
@@ -100,7 +100,27 @@ It's important to know that this policy does require the developer to manage the
...
+    }
+  }
+}
+```
+
+This policy works for both projects with and without custom native code. It works by using the `@expo/fingerprint` package to calculate a hash of your project during builds and updates to determine build-update compatibility (a.k.a. the runtime).
+
+### `"fingerprintExperimental"` runtime version policy
+
+> **warning** `fingerprintExperimental`is deprecated and will be removed in SDK 51.
```
Comment (amandeepmittal) [inline docs/pages/eas-update/runtime-versions.mdx:123]:
```suggestion
> **warning** `fingerprintExperimental`is deprecated and is removed from SDK 51 onwards.
```

Can we rephrase this as per the suggestion above so that we don't have to come back to this doc and update context in the warning to mention that "it has been removed" when we release SDK 51?

File: docs/pages/eas-update/runtime-versions.mdx [hunk 6]
```diff
@@ -100,7 +100,27 @@ It's important to know that this policy does require the developer to manage the
...
+{
+  "expo": {
+    "runtimeVersion": {
+      "policy": "fingerprint"
+    }
+  }
+}
+```
+
+This policy works for both projects with and without custom native code. It works by using the `@expo/fingerprint` package to calculate a hash of your project during builds and updates to determine build-update compatibility (a.k.a. the runtime).
```
Comment (amandeepmittal) [inline docs/pages/eas-update/runtime-versions.mdx:119]:
```suggestion
This policy works for both projects with and without custom native code. It works by using the `@expo/fingerprint` package to calculate the hash of your project during builds and updates to determine build-update compatibility (also known as the runtime).
```

Let's avoid using Latin abbreviations as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#abbreviations).

**Review by amandeepmittal — APPROVED** [review]
Left a few minor suggestions.
